src/methods/environment.js

```diff
@@ -130,7 +132,9 @@ export class EnvironmentImplementation {
       // a. If HasPrimitiveBase(V) is true, then
       if (this.HasPrimitiveBase(realm, V)) {
         // i. Assert: In this case, base will never be null or undefined.
-        invariant(base instanceof Value && !HasSomeCompatibleType(base, UndefinedValue, NullValue));
+        invariant(
```

Comment:
Rather than change the meaning of HasPrimitiveBase and relaxing this invariant, let's keep things as they are in the specification and introduce a separate branch where we handle pure functions.